fuel-vm/src/tests/predicate.rs

```diff
@@ -296,6 +296,138 @@ async fn predicate() {
+    );
+    // third input with own predicate
+    let predicate_2_index = 2;
+    let predicate_data_2_index = 3;
+
+    let input_3 = Input::coin_predicate_v2(
+        rng.r#gen(),
+        rng.r#gen(),
+        amount,
+        asset_id,
+        rng.r#gen(),
+        predicate_2_index,
+        predicate_data_2_index,
+    );
+    let predicate_bytes: Vec<_> = vec![op::ret(RegId::ONE)].into_iter().collect();
+
+    let predicate_static_witness = Witness::from(predicate_bytes);
+    let predicate_data_static_witness = Witness::from(vec![]);
+
+    let mut tx = TransactionBuilder::script_v2(script, script_data)
+        .add_input(input)
+        .add_input(input_2)
+        .add_input(input_3)
+        .add_output(Output::coin(rng.r#gen(), amount, asset_id))
+        .add_static_witness(predicate_static_witness.clone())
+        .add_static_witness(predicate_data_static_witness.clone())
+        .add_static_witness(predicate_static_witness)
+        .add_static_witness(predicate_data_static_witness)
+        .finalize();
+
+    assert_eq!(tx.inputs()[0].predicate_gas_used(), Some(0));
+    let result = tx.estimate_predicates(
+        &ConsensusParameters::standard().into(),
+        MemoryInstance::new(),
+        &EmptyStorage,
+    );
+
+    // Then
+    result.expect("Should estimate predicate");
+    assert_ne!(tx.inputs()[0].predicate_gas_used(), Some(0));
+}
+
+#[cfg(feature = "chargeable-tx-v2")]
+#[tokio::test]
+async fn script_v2__estimate_predicate__fails_if_predicate_not_found_in_static_witnesses()
+{
+    let mut rng = StdRng::seed_from_u64(2322u64);
+
+    // given
+    let script = vec![];
+    let script_data = vec![];
+    let predicate_index = 0;
+    let predicate_data_index = 1;
+    let asset_id = rng.r#gen();
+    let amount = rng.r#gen();
+    let input = Input::coin_predicate_v2(
+        rng.r#gen(),
+        rng.r#gen(),
+        amount,
+        asset_id,
+        rng.r#gen(),
+        predicate_index,
+        predicate_data_index,
+    );
+
+    let mut tx = TransactionBuilder::script_v2(script, script_data)
+        .add_input(input)
+        .add_output(Output::coin(rng.r#gen(), amount, asset_id))
+        .finalize();
+
+    // when
+    let result = tx.estimate_predicates(
+        &ConsensusParameters::standard().into(),
+        MemoryInstance::new(),
+        &EmptyStorage,
+    );
+
+    // Then
+    let err = result.expect_err("Should estimate predicate");
+    if let CheckError::PredicateVerificationFailed(
+        PredicateVerificationFailed::MissingPredicate {
+            input_indices,
+            predicate_index,
+        },
+    ) = &err
+    {
+        assert_eq!(predicate_index, &0);
+        assert_eq!(input_indices, &vec![0]);
+    } else {
+        panic!("Expected MissingPredicate error");
+    }
+}
+
+#[ignore]
+#[cfg(feature = "chargeable-tx-v2")]
+#[tokio::test]
+async fn script_v2__estimate_predicate__fails_if_predicate_data_not_found_in_static_witnesses()
+ {
+    todo!()
```

Comment:
Is this intentionally left as `todo!`?

Reply:
No. I just forgot to add this case. Thanks.